src/main/java/frc/robot/subsystems/Revolver.java

```diff
@@ -56,16 +42,22 @@ public void setAll(double drumSpeed, double feedWheelSpeed) {
     setFeedWheel(feedWheelSpeed);
   }
 
+  public void resetReps(){
+    m_drumReps = 0;
+    m_feedReps = 0;
+  }
+
   /**
    * Sets the speed of the drum
    * @param speed the speed to run the drum at as a percentage from -1 to 1
    */
   public void setDrum(double speed) {
     m_drumMotor.set(speed);
-    if (m_pdp.getCurrent(RobotMap.PDP.DRUM_PORT) > Constants.REVOLVER_DRUM_CURRENT_THRESHOLD) {
-      m_repsAboveDrum++;
-    } else {
-      resetDrumReps();
+    if (m_drumMotor.getEncoder().getVelocity()==0.0){
+      m_drumReps++;
+    }
+    if(m_drumReps > Constants.REVOLVER_REPS_THRESHOLD){
```

Comment:
I think we need additional logic here. The way I see it, we should have the following cases:

 - If the motor speed is 0, reset `m_drumReps` (Stopped)
 - If the motor speed is not 0 AND the velocity is not 0, reset `m_drumReps` (Rotating, all good)
 - If the motor speed is not 0 AND the velocity is 0, increment `m_drumReps` (Jammed)
 - If `m_drumReps` exceeds the threshold, schedule the `UnjamRevolver` and reset `m_drumReps`

Reply:
Actually, instead of checking if motor speed is not 0, we should only check if the revolver is rotating **forward**. Otherwise, if we get jammed while going backward, we'll get into an infinite loop of calling `UnjamRevolver`.